### PR TITLE
Infomation Lower Bound Search

### PR DESCRIPTION
**Pull request by ewfuentes**

### CODE REVIEW

File: experimental/beacon_sim/information_lower_bound_search.cc [hunk 1]
```diff
@@ -0,0 +1,150 @@
+
+#include "experimental/beacon_sim/information_lower_bound_search.hh"
```
Comment (ewfuentes) [inline experimental/beacon_sim/information_lower_bound_search.cc:2]: need to get rid of debugging prints

File: experimental/beacon_sim/information_lower_bound_search_test.cc [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    TestEnvironment env = create_triangle_test_env();
+    constexpr double GOAL_INFO_LOWER_BOUND = 1;
+    constexpr double START_INFO = 1;
+
+    // Action
+    const auto result = information_lower_bound_search(
+        env.road_map, env.start_idx, env.end_idx, START_INFO, GOAL_INFO_LOWER_BOUND, env.rev_prop);
+
+    // Verification
+    EXPECT_TRUE(false);
```
Comment (ewfuentes) [inline experimental/beacon_sim/information_lower_bound_search_test.cc:91]: Add real test criteria

File: experimental/beacon_sim/information_lower_bound_search_test.cc [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    TestEnvironment env = create_triangle_test_env();
+    constexpr double GOAL_INFO_LOWER_BOUND = 1;
+    constexpr double START_INFO = 10;
+
+    // Action
+    const auto result = information_lower_bound_search(
+        env.road_map, env.start_idx, env.end_idx, START_INFO, GOAL_INFO_LOWER_BOUND, env.rev_prop);
+
+    // Verification
+    EXPECT_TRUE(false);
```
Comment (ewfuentes) [inline experimental/beacon_sim/information_lower_bound_search_test.cc:105]: Add real test criteria

**Review by aknh9189 — APPROVED** [review]

File: experimental/beacon_sim/information_lower_bound_search.hh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    std::vector<int> path_to_goal;
+};
+
+InformationLowerBoundResult information_lower_bound_search(
+    const planning::RoadMap &road_map, const int start_idx, const int end_idx,
+    const double start_information, const double end_information_lower_bound,
+    const LowerBoundReversePropagator &propagator);
+
+namespace detail {
+struct InProgressPath {
```
[the remark this answers is not in the document]
Comment (ewfuentes) [reply experimental/beacon_sim/information_lower_bound_search.hh:31]: `InProgressPath` is meant to be an internal data structure that I'm only exposing for the sake of unit testing. I did want to separate the internal representation from the external API. I can easily imagine returning additional details about the search like the number of edges traversed, or the propagated bounds at each node as a part of the result.

File: experimental/beacon_sim/information_lower_bound_search.hh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    double info_lower_bound;
+    double cost_to_go;
+    std::vector<int> path_to_goal;
+
+    bool operator>(const InProgressPath &other) const { return cost_to_go > other.cost_to_go; }
+};
+
+struct MergeResult {
+    bool should_merge;
+    std::vector<int> to_boot;
```
[the remark this answers is not in the document]
Comment (ewfuentes) [reply experimental/beacon_sim/information_lower_bound_search.hh:41]: Each node has a list of paths from that node to the goal node, along with the information lower bound required for that path and the cost of that path. Given a new path to a node, `should_merge()` returns:
a) If that new path should be added to the list of paths maintained by that node, and
b) If that new path should be added, which paths, if any, should be dropped from the list of paths maintained by the node. This is represented by the vector of indices that should be dropped or booted, hence `to_boot`.

I'll add some comments :).

**Pull request merged**